core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -505,21 +488,28 @@ public boolean mkdirs(Path cPath, FsPermission permission) throws IOException {
   /**
    * Attempts to open the specified file for reading.
    *
-   * @param cPath the file name to open
+   * @param path the file name to open
    * @param bufferSize the size in bytes of the buffer to be used
    * @return an {@link FSDataInputStream} at the indicated path of a file
    * @throws IOException if the file cannot be opened (e.g., the path is a folder)
    */
   @Override
-  public FSDataInputStream open(Path cPath, int bufferSize) throws IOException {
-    LOG.info("open({}, {})", cPath, bufferSize);
+  public FSDataInputStream open(Path path, int bufferSize) throws IOException {
+    LOG.info("open({}, {})", path, bufferSize);
     if (mStatistics != null) {
       mStatistics.incrementReadOps(1);
     }
 
-    AlluxioURI path = new AlluxioURI(HadoopUtils.getPathWithoutScheme(cPath));
-    return new FSDataInputStream(new HdfsFileInputStream(path, HadoopUtils.getHDFSPath(path,
-            mUnderFSAddress), getConf(), bufferSize, mStatistics));
+    AlluxioURI uri = new AlluxioURI(HadoopUtils.getPathWithoutScheme(path));
+    URIStatus status;
+    try {
+      status = mFileSystem.getStatus(uri);
+    } catch (AlluxioException e) {
+      throw new IOException(e);
+    }
+    return new FSDataInputStream(
+        new HdfsFileInputStream(uri, new Path(status.getUfsPath()), getConf(), bufferSize,
+            mStatistics));
```

Comment:
I think we can just remove the second argument and compute it in the constructor of `HdfsFileInputStream`?